src/swupd_lib/statedir.c

```diff
@@ -56,44 +66,128 @@ char *statedir_get_tracking_file(const char *bundle_name)
 	return sys_path_join("%s/%s/%s", globals.state_dir, TRACKING_DIR, bundle_name);
 }
 
+char *statedir_get_telemetry_record(char *record)
+{
+	return sys_path_join("%s/%s/%s", globals.state_dir, TELEMETRY_DIR, record);
+}
+
+char *statedir_get_swupd_lock(void)
+{
+	return sys_path_join("%s/%s", globals.state_dir, LOCK);
+}
+
+char *statedir_get_version(void)
+{
+	return sys_path_join("%s/%s", globals.state_dir, VERSION_FILE);
+}
+
+/* ************* */
+/* Cache section */
+/* ************* */
+
+static char *get_url(void)
+{
+	size_t i, j = 0;
+	char *url = strdup_or_die(globals.content_url);
+	char *converted_url = malloc_or_die(str_len(url) + 1);
```

Comment:
It's better to create another function `convert_url(const char *url)` so it can be possibly reused for other urls